ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAutoML.scala

```diff
@@ -102,9 +102,9 @@ class H2OAutoML(override val uid: String) extends Estimator[H2OMOJOModel]
         " your 'excludeAlgo', 'maxModels' or 'maxRuntimeSecs' properties.") with NoStackTrace
     }
 
-    val binaryModel = aml.leader()
-    val mojoData = ModelSerializationSupport.getMojoData(binaryModel)
-    val model = H2OMOJOModel.createFromMojo(mojoData, Identifiable.randomUID(aml.leader()._parms.algoName()))
+    binaryModel = Some(aml.leader())
+    val mojoData = ModelSerializationSupport.getMojoData(binaryModel.get)
```

Comment:
Can we avoid usage of `get` method and introduce a local variable that is later encapsulated into the option assigned to `binaryModel`?

Reply:
I was also thinking of instead of having binaryModel as Option, use BaseModel directly and use null instead of None. That would solve it, however I like options a bit more then using null